lightning/src/ln/channelmanager.rs

```diff
@@ -6505,10 +6702,16 @@ impl<'a, Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 		// pending_outbound_payments_no_retry is for compatibility with 0.0.101 clients.
 		let mut pending_outbound_payments_no_retry: Option<HashMap<PaymentId, HashSet<[u8; 32]>>> = None;
 		let mut pending_outbound_payments = None;
+		let mut phantom_scid_namespace: Option<fake_scid::Namespace> = None;
 		read_tlv_fields!(reader, {
 			(1, pending_outbound_payments_no_retry, option),
 			(3, pending_outbound_payments, option),
+			(5, phantom_scid_namespace, option),
 		});
+		if phantom_scid_namespace.is_none() {
+			phantom_scid_namespace = Some(fake_scid::Namespace::phantom(&args.keys_manager));
```

Comment:
Shouldn't we store a `u8` offset instead of picking a value for phantoms specifically. Instead of storing a value for each namespace, we store one "offset" and then when we want to know a namespace's value, we just add the offset and mod by `MAX_NAMESPACES`.

Reply:
Do you mean we should store the 32-byte random value that we got the namespace from (in `fake_scid::Namespace::phantom`), and use it for all namespaces?

Reply:
I think we should just store the 3-bit modulo value, no? Like, instead of phantom namespace being a specific value, we define phantom to be 0 and others to increment by 1, with a random offset for the whole thing (mod 8)